les02.js

```diff
@@ -0,0 +1,165 @@
+
+var a = 2;
+var x = 1 + (a *= 2);
+
+//Задание 3
+
+var a = +prompt('Введите первое целое число');
+var b = +prompt('Введите второе целое число');
+
+if(a >= 0 && b>=0) {
+    alert('Разность чисел ' + (a - b));
+} else if(a < 0 && b < 0) {
+    alert('Произведение чисел ' + (a * b));
+} else {
+    alert('Сумма чисел  ' + (a + b));
+}
+
+//Задание 4
+
+var a = +prompt('Введите число от 0 до 15');
+var c = '';
+switch(a) { 
+    case 0:
+      c = c + '0 ';   
+    case 1:
+      c = c + '1 ';
+    case 2:
+      c = c + '2 ';
+    case 3:
+      c = c + '3 ';
+    case 4:
+      c = c + '4 ';
+    case 5:
+      c = c + '5 ';
+    case 6:
+      c = c + '6 ';
+    case 7:
+      c = c + '7 ';
+    case 8:
+      c = c + '8 ';
+    case 8:
+      c = c + '8 ';
+    case 9:
+      c = c + '9 ';
+    case 10:
+      c = c + '10 ';
+    case 11:
+      c = c + '11 ';
+    case 12:
+      c = c + '12 ';
+    case 13:
+      c = c + '13 ';
+    case 14:
+      c = c + '14 ';
+    case 15:
+      c = c + '15 ';
+      break;
+    default:
+      c = c + 'Число не верно';
+      break;
+}
+
+alert(c);
+
+//Задание 5
+
+function plus(a, b) {
+    return a + b;
+}
+function minus(a, b) {
+    return a - b;
+}
+function umnozh(a, b) {
+    return a * b;
+}
+function delit(a, b) {
+    return a / b;
+}
+
+var a = +prompt('Введите первое число');
+var b = +prompt('Введите второе число');
+
+alert('Сложение: ' + plus(a, b));
+alert('Вычитание: ' + minus(a, b));
+alert('Умножение: ' + umnozh(a, b));
+alert('Деление: ' + delit(a, b));
+
+//Задание 6
+
+function plus(a, b) {
+    return a + b;
+}
+function minus(a, b) {
+    return a - b;
+}
+function umnozh(a, b) {
+    return a * b;
+}
+function delit(a, b) {
+    return a / b;
+}
+function mathOperation(arg1, arg2, operation) {
+    switch(operation) {
+        case 'plus':
+            return plus(arg1,arg2);
+            break;
+        case 'minus':
+            return minus(arg1,arg2);
+            break;
+        case 'delit':
+            return delit(arg1,arg2);
+            break;
+        case 'umnozh':
+            return umnozh(arg1,arg2);
+            break;
+        default:
+            alert('Операция указана неверно');
+            break;
+    }
+}
+
+var a = +prompt('Введите первое число');
+var b = +prompt('Введите второе число');
+var oper = prompt('Введите операцию');
+
+alert(mathOperation(a, b, oper));
+
+/*Задание 7
+Сравнить null и 0. Обьяснить результат*/
+
+null > 0    //false
+null < 0    //false
+null == 0   //false
+null >= 0   //true
+null <= 0   //true
+
+/* Насколько я понял, результаты с первого по третий связаны с механизмом абстрактного алгоритма сравнения неравенств и равенств. А операторы ">=" и "<=" работают по принципу: если null > 0 принимает значение FALSE, то null <= 0 принимает значение TRUE.*/
+
+//Задание 8
+
+function power(val, pow) {
+    if (pow === 0) {
+        return 1;
+    }
+    
+    return val * power(val, pow-- - 1);
```

Comment:
Зачем здесь декремент?
```suggestion
    return val * power(val, pow - 1);
```